src/k8s/pkg/k8s/cluster/cluster.go

```diff
@@ -0,0 +1,170 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/microcluster/client"
+	"github.com/canonical/microcluster/microcluster"
+)
+
+// Bootstrap sets up new cluster and returns the information about the daemon.
+func Bootstrap(ctx context.Context, opts ClusterOpts) (ClusterMember, error) {
```

Comment:
I prefer returning a `*ClusterMember` instead, to minimize `copy`ing things around.